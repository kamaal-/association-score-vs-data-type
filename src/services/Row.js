import {initialDataTypes} from '../config/'

export default class Row {
    constructor(data){
        this.data = data 
        this.setupDataTypes()
    }

    setupDataTypes(data){
        const self = this
        this.dataTypeKeyValues = (Object.keys(initialDataTypes)|| []).map( key => {
            return {...initialDataTypes[key], value: self.data.association_score.datatypes[key]}
        })
    }

    get geneId(){
        return this.data.target.id
    }

    get symbol(){
        return this.data.target.gene_info.symbol
    }

    get name(){
        return this.data.target.gene_info.name
    }

    get overallScore(){
        return this.data.association_score.overall
    }

    get visualizerData(){
        return this.dataTypeKeyValues
    }

    set dataType(dataType){
        this.dataTypeKeyValues.push(dataType)
    }
}