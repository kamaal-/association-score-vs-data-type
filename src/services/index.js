import Axios from "axios"
import Row from './Row'
import {totalItemsFromTo} from '../config/'

export const dataToRow = d => new Row(d)

export const transformDataToRows  = data => ((data && data.data) || [])
    .map(dataToRow)
    .sort(sortRowByOverallScore)
    .slice(...totalItemsFromTo)

export const sortRowByOverallScore = (a, b) => {
    if(a.overallScore > b.overallScore) return -1;
    if(a.overallScore < b.overallScore) return 1;
    return 0;
}

export const getInitialData = (cb) => {
    Axios.get('https://demo6922545.mockable.io/')
        .then(({data}) => cb(transformDataToRows (data)))
        .catch(e => {
            console.log(e)
        })
}