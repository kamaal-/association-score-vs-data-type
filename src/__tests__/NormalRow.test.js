import React from 'react';
import { render, waitForElement, fireEvent } from '@testing-library/react';
import Row from '../components/Rows/NormalRow';
import data from '../config/mock.data';
import { transformDataToRows, Â } from '../services/'
import ReactDOM from 'react-dom'

describe('Row', () => {

    const setup = (fn = () => {}) => {
        const container = document.createElement('tbody')
        const rows = transformDataToRows(data())	
        return render(<Row row={rows[0]} index={0} collapsedRow={2} toggleCollapsedRow={fn}/>, {
            container: document.body.appendChild(container)
        });
    }

  	it('renders without crashing', () => {
        const div = document.createElement('tbody');
        const rows = transformDataToRows(data())	
    	ReactDOM.render(<Row row={rows[0]} index={0} collapsedRow={2} toggleCollapsedRow={() => {}}/>, div);
    	ReactDOM.unmountComponentAtNode(div);
    });

    it('should trigger parent event', async (done) => {
        const setNewIndexAtParent= jest.fn();
        const {getByTestId} = setup(setNewIndexAtParent)
        const el = await waitForElement(() => getByTestId('collapse-btn'))
        fireEvent.click(el)
        expect(setNewIndexAtParent).toBeCalled()
        done()
    })

  	it('Should render 1 rows', () => {
		const { getAllByTestId } = setup()
		expect( getAllByTestId('actual-tr').length).toBe(1);
    });
      
    it('should match snapshot', () => {
        const { asFragment } = setup()
        expect(asFragment()).toMatchSnapshot();
    })

})
 