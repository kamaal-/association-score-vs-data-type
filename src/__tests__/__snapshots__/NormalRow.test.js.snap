// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row should match snapshot 1`] = `
<DocumentFragment>
  <tr
    class="MuiTableRow-root"
    data-testid="actual-tr"
  >
    <th
      class="MuiTableCell-root"
      scope="row"
      small="small"
    >
      <button
        aria-label="delete"
        class="MuiButtonBase-root MuiIconButton-root"
        data-testid="collapse-btn"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </th>
    <td
      class="MuiTableCell-root MuiTableCell-alignLeft"
    >
      EGFR
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-alignLeft"
    >
      ENSG00000146648
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-alignLeft"
    >
      epidermal growth factor receptor
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-alignLeft"
    >
      1
    </td>
  </tr>
</DocumentFragment>
`;
